### PR TITLE
Improve locking failures output

### PR DESCRIPTION
**Pull request by ljacomet**
By using `ResolutionProvider`, advice and documentation link properly appear in the
Try section on the console output.

~~Still plan to add a release not entry, in parallel of CI confirming I did not break anything.~~ Release note added

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -63,6 +63,32 @@ A new [`named(Spec<String>)` method](javadoc/org/gradle/api/NamedDomainObjectCol
 [`ModuleDependencyCapabilitiesHandler#requireCapability(Object)`](javadoc/org/gradle/api/artifacts/ModuleDependencyCapabilitiesHandler.html#requireCapability-java.lang.Object-),
 and [`CapabilitiesResolution#withCapability(Object, Action)`](javadoc/org/gradle/api/artifacts/CapabilitiesResolution.html#withCapability-java.lang.Object-org.gradle.api.Action-).
 
+### Error and warning reporting improvements
+
+Gradle provides a rich set of error and warning messages to help you understand and resolve problems in your build.
+
+#### Dependency locking now separates the error from the possible action to try
+
+[Dependency locking](userguide/dependency_locking.html) is a mechanism for ensuring reproducible builds also when using dynamic dependency versions.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:72]:
```suggestion
[Dependency locking](userguide/dependency_locking.html) is a mechanism for ensuring reproducible builds when using dynamic dependency versions.
```

**Review by lkasso — APPROVED** [review]
Docs ok. Minor grammar suggestion.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -63,6 +63,32 @@ A new [`named(Spec<String>)` method](javadoc/org/gradle/api/NamedDomainObjectCol
...
+This release improves error messages by separating the error from the possible action to fix the issue in the console output.
+Errors from invalid [lock file format](userguide/dependency_locking.html#lock_state_location_and_format) or [missing lock state when strict mode is enabled](userguide/dependency_locking.html#fine_tuning_dependency_locking_behaviour_with_lock_mode) are now displayed as illustrated below:
+
+```
+FAILURE: Build failed with an exception.
+
+* What went wrong:
+Execution failed for task ':dependencies'.
+> Could not resolve all dependencies for configuration ':lockedConf'.
+   > Invalid lock state for lock file specified in '<projec>/lock.file'. Line: '<<<<<<< HEAD'
```
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md:83]:
```suggestion
   > Invalid lock state for lock file specified in '<project>/lock.file'. Line: '<<<<<<< HEAD'
```

**Review by big-guy — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle merge

**Pull request merged**